src/templater.rs

```diff
@@ -543,18 +544,34 @@ where
 pub struct CommitOrChangeId<'a> {
     repo: RepoRef<'a>,
     id_bytes: Vec<u8>,
+    is_commit_id: bool,
 }
 
 impl<'a> CommitOrChangeId<'a> {
-    pub fn new(repo: RepoRef<'a>, id: &impl ObjectId) -> Self {
+    pub fn commit_id(repo: RepoRef<'a>, id: &CommitId) -> Self {
         CommitOrChangeId {
             repo,
             id_bytes: id.to_bytes(),
+            is_commit_id: true,
+        }
+    }
+
+    pub fn change_id(repo: RepoRef<'a>, id: &ChangeId) -> Self {
+        CommitOrChangeId {
+            repo,
+            id_bytes: id.to_bytes(),
+            is_commit_id: false,
         }
     }
 
     pub fn hex(&self) -> String {
-        hex::encode(&self.id_bytes)
+        if self.is_commit_id {
+            hex::encode(&self.id_bytes)
+        } else {
+            // TODO: We can avoid the unwrap() and make this more efficient by converting
+            // straight from bytes.
+            to_reverse_hex(&hex::encode(&self.id_bytes)).unwrap()
```

Comment:
Maybe better to migrate this to `ChangeId::hex()` and `::from_hex()` because the reverse hex is a canonical form of change id, but I don't think such change is needed at this point.